src/app/store/trips/selectors.ts

```diff
@@ -0,0 +1,69 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { Trip } from '../../models/trips';
+import { TripState, tripsFeatureKey } from './reducer';
+
+export const selectFeature = createFeatureSelector<TripState>(tripsFeatureKey);
+
+const deriveDatesAndCost = (trip: Trip) => {
+  let startDate = new Date(8640000000000000);
+  let endDate = new Date(-8640000000000000);
+  let totalTripCost = 0;
+
+  if (trip.itinerary && trip.itinerary.length > 0) {
+    trip.itinerary.forEach(activity => {
+      const newStartDate = activity.startDate.toDate();
+      const newEndDate = activity.endDate.toDate();
+
+      startDate = startDate
+        ? newStartDate < startDate
+          ? newStartDate
+          : startDate
+        : newStartDate;
+      endDate = endDate
+        ? newEndDate > endDate
+          ? newEndDate
+          : endDate
+        : newEndDate;
```

Comment:
this format is very confusing

Reply:
It is. Sadly Prettier is fighting me on this. So I am going to leave it as is.

Reply:
Just turn it into real if statements. Set it to the fallback, then overwrite it if the thing exists.